neuro_san/internals/run_context/langchain/core/base_tool_factory.py

```diff
@@ -0,0 +1,267 @@
+
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Set
+from typing import Union
+
+from logging import Logger
+from logging import getLogger
+
+from langchain_core.tools.base import BaseTool
+
+from neuro_san.internals.interfaces.async_agent_session_factory import AsyncAgentSessionFactory
+from neuro_san.internals.interfaces.context_type_toolbox_factory import ContextTypeToolboxFactory
+from neuro_san.internals.interfaces.invocation_context import InvocationContext
+from neuro_san.internals.journals.journal import Journal
+from neuro_san.internals.messages.agent_message import AgentMessage
+from neuro_san.internals.run_context.interfaces.agent_network_inspector import AgentNetworkInspector
+from neuro_san.internals.run_context.interfaces.tool_caller import ToolCaller
+from neuro_san.internals.run_context.langchain.core.langchain_openai_function_tool \
+    import LangChainOpenAIFunctionTool
+from neuro_san.internals.run_context.langchain.mcp.langchain_mcp_adapter import LangChainMcpAdapter
+from neuro_san.internals.run_context.utils.external_agent_parsing import ExternalAgentParsing
+from neuro_san.internals.run_context.utils.external_tool_adapter import ExternalToolAdapter
+
+
+class BaseToolFactory:
+    """
+    Creates langchain BaseTools.
+    """
```

Comment:
Factor out all the methods that are used to create BaseTools from LangChainRunContext.
LangChainRunContext was getting too big (nearing 1000 lines).

It's possible that this class could be broken up further, but not in scope for this PR.